app/controllers/users/sessions_controller.rb

```diff
@@ -2,36 +2,38 @@ module Users
   class SessionsController < Devise::SessionsController
     include ::ActionView::Helpers::DateHelper
 
-    prepend_before_action :skip_timeout, only: [:active]
     skip_before_action :session_expires_at, only: [:active]
 
-    def skip_timeout
-      request.env['devise.skip_trackable'] = true
-    end
-
-    def new
-      super
-    end
-
     def create
       track_authentication_attempt(params[:user][:email])
       super
     end
 
     def active
       response.headers['Etag'] = '' # clear etags to prevent caching
-      render json: { live: current_user.present?, timeout: session[:session_expires_at] }
+      Rails.logger.debug("alive?:#{alive?} expires_at:#{expires_at} now:#{Time.zone.now}")
+      render json: { live: alive?, timeout: expires_at }
     end
 
     def timeout
       analytics.track_anonymous_event('Session Timed Out')
-
       flash[:timeout] = t('session_timedout')
+      sign_out current_user
```

Comment:
I changed to `sign_out` with no argument. IIRC I had not done that originally because I had the order wrong: must sign out first, then set the flash message, since the flash is tied to the (now new) session.